packages/mermaid/src/diagrams/xychart/parser/xychart.jison

```diff
@@ -57,6 +57,7 @@
 
 "["                                       return 'SQUARE_BRACES_START'
 "]"                                       return 'SQUARE_BRACES_END'
+"NaN"                                     return 'NAN';
```

Comment:
NaN is rather JS-originated, while many programming languages use their own keywords. Why not simply omitting the value?